### PR TITLE
Disabled tinyfiledialogs when running headless

### LINKED ISSUE #15965: Disable tinyfiledialog popups when running in headless mode
Currently we still pop up tinyfiledialogs in headless mode, which we shouldn't. This would fix #15962.

### PR DESCRIPTION
**Pull request by asajeffrey**
Disable tinyfiledialogs when running in headless mode.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15965 and #15962.
- [X] These changes do not require tests because we don't test tinyfiledialog pop-ups.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15967)

### CODE REVIEW

**Comment by asajeffrey** [comment]
cc @jdm

File: components/net/filemanager_thread.rs [hunk 1]
```diff
@@ -557,7 +558,8 @@ impl FileManagerStore {
 
 fn select_files_pref_enabled() -> bool {
     PREFS.get("dom.testing.htmlinputelement.select_files.enabled")
-         .as_boolean().unwrap_or(false)
+         .as_boolean().unwrap_or(false) &&
+         !opts::get().headless
```
Comment (asajeffrey) [inline components/net/filemanager_thread.rs]: Oops, got the semantics of this flag inverted.

**Comment by asajeffrey** [comment]
The unit tests were failing because the UI provider wasn't being called in headless mode. The fix is to move the headless check inside the UI provider.

**Comment by jdm** [comment]
Looks good to merge after squashing.

**Comment by asajeffrey** [comment]
Squashed. @bors-servo r=jdm

**Pull request merged**